client/post-editor/editor-sharing/index.jsx

```diff
@@ -1,85 +1,24 @@
 /**
  * External dependencies
  */
-var React = require( 'react' );
+import React, { PropTypes } from 'react';
 
 /**
  * Internal dependencies
  */
-var PublicizeOptions = require( './publicize-options' ),
-	SharingLikeOptions = require( './sharing-like-options' ),
-	postUtils = require( 'lib/posts/utils' );
-
-module.exports = React.createClass( {
-	displayName: 'EditorSharing',
-
-	propTypes: {
-		site: React.PropTypes.object,
-		post: React.PropTypes.object,
-		isNew: React.PropTypes.bool,
-		connections: React.PropTypes.array,
-		fetchConnections: React.PropTypes.func
-	},
-
-	isSharingButtonsEnabled() {
-		if ( ! this.props.site || ! this.props.site.jetpack ) {
-			return true;
-		}
-
-		if ( ! this.props.site.isModuleActive( 'sharedaddy' ) ) {
-			return false;
-		}
-
-		return true;
-	},
-
-	isLikesEnabled() {
-		if ( ! this.props.site || ! this.props.site.jetpack ) {
-			return true;
-		}
-
-		if ( ! this.props.site.isModuleActive( 'likes' ) ) {
-			return false;
-		}
-
-		return true;
-	},
-
-	renderSharingLikeOptions() {
-		if ( ! this.isSharingButtonsEnabled() && ! this.isLikesEnabled() ) {
-			return null;
-		}
-
-		return(
-			<SharingLikeOptions
-				post={ this.props.post }
-				site={ this.props.site }
-				isSharingButtonsEnabled={ this.isSharingButtonsEnabled() }
-				isLikesEnabled={ this.isLikesEnabled() }
-				isNew={ this.props.isNew } />
-		);
-	},
-
-	renderPublicizeOptions() {
-		if ( postUtils.isPage( this.props.post ) ) {
-			return;
-		}
-
-		return (
-			<PublicizeOptions
-				post={ this.props.post }
-				site={ this.props.site }
-				connections={ this.props.connections }
-				fetchConnections={ this.props.fetchConnections } />
-		);
-	},
-
-	render: function() {
-		return (
-			<div className="editor-sharing">
-				{ this.renderPublicizeOptions() }
-				{ this.renderSharingLikeOptions() }
-			</div>
-		);
-	}
-} );
+import PublicizeOptions from './publicize-options';
+import SharingLikeOptions from './sharing-like-options';
+
+export default function EditorSharing( { post, site } ) {
+	return (
+		<div className="editor-sharing">
+			<PublicizeOptions post={ post } site={ site } />
```

Comment:
>A wise PR reviewer once told me to use the spread operator in spots like this `{ ...{ post, site } }` 😉 

Aha! 😆  I do think there's a bit of a balance here in terms of readability. How it's written here is still pretty clear without too much repetition or line length. My arbitrary personal minimum is 3 before I'd consider the spread.

That said, there's a third way this could be written that would take advantage of the fact that we're basically just passing along props from the parent:

```jsx
export default function EditorSharing( props ) {
	return (
		<div className="editor-sharing">
			<PublicizeOptions { ...props } />
			<SharingLikeOptions { ...props } />
		</div>
	);
}
```

But here you'd lose some transparency into what exactly is passed (until you examine `propTypes` anyways).